### PR TITLE
refs: adds desk refs

### PR DESCRIPTION
**Pull request by nerveharp**
- Ports over docket state from garden
- Adds an AppReference component
- Changes the types to reflect changes to desk cites

image

fixes tloncorp/homestead#612

### CODE REVIEW

File: ui/src/components/References/AppReference.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+interface AppReferenceProps {
+  desk: string;
+}
+
+export default function AppReference({ desk }: AppReferenceProps) {
+  const [ship, deskId] = desk.split('/');
+  const treaty = useTreaty(ship, deskId);
+  const calm = useCalm();
+  const href = `/apps/grid/leap/search/${ship}/apps/${ship}/${deskId}`;
+
```
Comment (arthyn) [inline ui/src/components/References/AppReference.tsx:16]: missing a call for treaties we don't have:

```
  useEffect(() => {
    if (!treaty) {
      useDocketState.getState().requestTreaty(ship, desk);
    }
  }, [treaty, ship, desk]);
  ```

**Review by arthyn — CHANGES_REQUESTED** [review]
Just missing one piece, but this is great!

**Comment by liam-fitzgerald** [comment]
@nerveharp what's going on with CI?

**Comment by nerveharp** [comment]
> @nerveharp what's going on with CI?

Test for ChatMessage is borked, @arthyn is looking into it. I'm lost

**Review by arthyn — APPROVED** [review]
works now!

**Pull request merged**